### ISSUE TITLE
Handling alternative transcription systems: American PA as an example

### ISSUE THREAD

**Issue by LinguList** [glottobank/clpa#45]
* https://en.m.wikipedia.org/wiki/Americanist_phonetic_notation

They have lots of symbols we know well and may also alias in the clts, but we may think of a principled way to handle this. I think, the classical IPA descriptions for sounds, which now follow the schema to list a **base** description (*phonation*, *place*, *manner*) in a tsv-table, plus additional derivations following in the EXTRA column, encoded as `feature:value`, separated by comma, should enable us to link different transcription systems to the same feature bundle. The *generated* assessment, which will be another attribute of all subclasses of the main class `Sound`, indicates whether we have the very sound in our system in this explicit form, or whether we automatically added features based on diacritics to left or right of the character match (using simple regex syntax), should probably reserved to *ipalike* phonetic notations. 

When adding alternative transcription systems, like the APA of Americanists, or historical systems, we need to decide whether we handle those in orthography profiles, or whether it is enough to add them as aliases, maybe with an additional tag, showing in which transcription system they are usually used. Orthoprofiles have the disadvantage that they can be used to make a secondary segmentation:

Grapheme | Segments | Description
--- | --- | ---
á | a ⁵ | x y z vowel, high tone

But `clts` should only be applied after segmentation, so I'd favour an alias annotation, maybe even allowing to specify the base alphabet, like:

Grapheme | Description | Extra | Transcription_System
--- | --- | --- 
č | unvoiced postalveolar affricate | | APA
tʃ | unvoiced postalveolar affricate | | IPA

By calling the CLTS class, we could then add a keyword specifying the base notation, which defaults to "ipa", but if set to "None" would give preference to "ipa" but otherwise guess and treat the rest as an alias.

```
>>> clts = CLTS()
>>> sound = clts("tʃ", notation="ipa")
>>> sound = clts("č", notation="apa")
>>> sound = clts("č", notation=None) 
```
Alternatively, this could already be handled upon instantiation:

```
>>> clts = CLTS(notation="ipa")
```

This may even be simpler, as the whole instantiation of the data could be done in a rush, and preference for interpretation in case of ambiguous symbols (especially: how to *write* the stuff in normalized form) would give preference to the notation system specified, if it is found (a notation system may well lack a sound).

Problem is that we would be forced to handle probably many different notation systems encoding the same sound. I guess the simplest would be to handle notation as a separated list, like:

Grapheme | Description | Extra | Transcription_System
--- | --- | --- 
č | unvoiced postalveolar affricate | | apa
tʃ | unvoiced postalveolar affricate | | ipa, phoible, pbase

Adding multiple lines for different notation systems would blow up the number of rows and make the data unhandable (so far it is still okay). 

A problem in writing sounds (each sound as the attributes `name` and `__str__`), are ambiguities, such as we have in IPA, with the various annotations for `[`tʃ`]`. So we would also be forced to handle preference in writing the string, which probably needs to be annotated somehow in an explicit way. Not sure how to handle this, but having the possibility to actually link across transcription systems and to write the sounds in a specific transcription system would be a good feature.

**Comment by xrotwang** [comment]
Any chance to define preference of symbols for the same sound based on
unicode properties, i.e. shortest unicode string or similar?
Am 28.03.2017 08:09 schrieb "Johann-Mattis List" <notifications@github.com>:

**Comment by xrotwang** [comment]
I like the approach of different base notations, and would use the analogy
to number systems to model the API.
Am 28.03.2017 08:09 schrieb "Johann-Mattis List" <notifications@github.com>:

**Comment by xrotwang** [comment]
To elaborate on my last point: It would be nice to have `sound` available as a type and factory function just like `int`:
```python
>>> int('12', base=8)
10
>>> sound("tʃ", notation="ipa")
"tʃ"
>>> isinstance(_, sound)
True
```

**Comment by LinguList** [comment]
> Any chance to define preference of symbols for the same sound based on unicode properties, i.e. shortest unicode string or similar?

well, not, as I'd prefer "tʃ" over the unified one, so this should be explicit, I am afraid, and we'd need additional markup to say, x is alias of IPA, but base of IPA is whatever...

**Comment by SimonGreenhill** [comment]
This would be quite cool:

```python
sound("ʤ", notation="ipa") == sound("j", notation="asjp")
```

**Comment by LinguList** [comment]
yes, I even forgot that: we'd have "asjp" (even with modifiers!!!), also "gld" (=tower of babel phonetic notation), etc.

**Comment by LinguList** [comment]
So, the more I think about it: if a system, like, say, APA fails to have a sound, and cannot generate it, it would just have return our UnknownSound instance, which I already added. The generation principle is quite powerful, but we'd need to see how well we can handle asjp, etc. with that, so we may better just spell those sounds out.

The more I think, we should separate each phonetic notation to a single file, so we'd have:

* clts (our preferred system, which would be generalized and allow for aliases to make it easier to match sounds, so ž would be rendered as ʤ, etc.
* ipa (following the official style guide, no links to other systems)
* gld (tower of babel system)
* asjp
* apa (the example mentioned above, maybe only one table for illustration)
* phoible
* pbase (Mielke's database)
* fonetikode (the link between phoible and Ruhlen's database by Scott Moisik and Dan Dediu)

Translation between sounds would take the full base letter specification "voiced bilabial plosive" to link between systems, so this is our "concept set", to have a tertium comparationis, a comparison sound.

Every system can define aliases, with things not labeled as alias being the default for the Sound.__str__. We still need to distinguish the following major sound types:

* Consonant
* Vowel
* Diphthong
* Click
* Tone
* Marker (specific markup, not interpreted as sound)
* Cluster (class currently lacking, pointing to things like double-articulation, e.g., `[`pt`]`, etc.)

In addition to the phoible-consonants.tsv, each system would have a *-diacritics.tsv, indicating additional features to be added to the left or right of a base sound so that we can generate the sound if it is not in our database.

A sound could be initiated by either reading the description:

```python
sound('voiced bilabial plosive', notation='ipa')
```

and by reading the character:

```python
sound('b')
```
